Fix typos, improve wording in a number of different places

jhoak opened the pull request.

mtbentley commented on organizational/linkstate.rst:
```diff
@@ -1,19 +1,19 @@
 - They will hold the whip on getting members to invite speakers.
 - They will evaluate talk proposals.
-- They are the person who will finalize the schedule: determine time slots, do
+- Ultimately, they will finalize the schedule by determining time slots, doing
```
`doing` sounds awkward

jhoak replied:
"handling the logistics" then? (edit to lower bloat) or managing?

mtbentley replied:
handling or managing would work now that you mention it

mtbentley commented:
Thanks for these changes!  I'll leave it open for other people to comment and for changes to be made, but this is a much appreciated update.

jhoak commented on organizational/linkstate.rst:
```diff
@@ -1,19 +1,19 @@
 - They will hold the whip on getting members to invite speakers.
 - They will evaluate talk proposals.
-- They are the person who will finalize the schedule: determine time slots, do
+- Ultimately, they will finalize the schedule by determining time slots, doing
```
managing?

jhoak commented:
Updated

jhoak commented:
Scanned thru the changes once more and found I needed to add length to a couple headers -- that's done in the latest commit (sorry for the spam)

brenns10 commented on organizational/slack.rst:
```diff
@@ -1,12 +1,12 @@
-`Hacsoc Slack`_ 
+`Click here to go to the official HacSoc Slack.`_ 
 
 .. _Hacsoc Slack: https://hacsoc.slack.com
```
This change will almost certainly break the link. Notice how the reference on line 6 matches the link text on line 4. You can either make them match again, or switch to the inline link syntax:

    `link text <link target>`_

[Quick reference](http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets). Be sure to check these things with a `make html` - there will be "compiler warnings" for broken links, etc.

mtbentley replied:
Hmm it seems that we should add Travis to check this automagically?

brenns10 replied:
@matthewbentley !! *do not* enable Travis yet

brenns10 replied:
If you enable it for pull requests, then it will update the site

brenns10 replied:
I'll need to update `push.sh` to check whether it is within a PR (just an environment variable check, NBD), but we can't switch it yet

mtbentley replied:
Yeah I won't change anything yet.  Just a suggestion for the future.

jhoak commented:
Subsections broke again (a git mistake on my end removed the extra lines I added in an earlier commit) so I fixed those again. Everything should be 👌 now
